apiserver/facades/client/imagemetadatamanager/package_test.go

```diff
@@ -34,24 +37,37 @@ type baseImageMetadataSuite struct {
 
 	api   *imagemetadatamanager.API
 	state *mockState
+
+	modelConfigService *MockModelConfigService
+	ctrl               *gomock.Controller
 }
 
 func (s *baseImageMetadataSuite) SetUpSuite(c *gc.C) {
 	s.BaseSuite.SetUpSuite(c)
 	imagetesting.PatchOfficialDataSources(&s.CleanupSuite, "test:")
 }
 
+func (s *baseImageMetadataSuite) setupMocks(c *gc.C) {
+	ctrl := gomock.NewController(c)
+	s.modelConfigService = NewMockModelConfigService(ctrl)
+	s.ctrl = ctrl
+}
+
 func (s *baseImageMetadataSuite) SetUpTest(c *gc.C) {
 	s.BaseSuite.SetUpTest(c)
 	s.resources = common.NewResources()
 	s.authorizer = testing.FakeAuthorizer{Tag: names.NewUserTag("testuser"), Controller: true, AdminTag: names.NewUserTag("testuser")}
 
 	s.state = s.constructState(testConfig(c))
 
+	// Need to create the mock modelConfigService first so we can pass it to the API
+	s.setupMocks(c)
```

Comment:
You can't call this here, it's because the `gc.C` isn't the same one. You'll need to call this for every function.

```go
defer s.setupMocks(c).Finish()
```